File: testdata/vulnerable.java
import java.security.MessageDigest;
import javax.crypto.Cipher;

public class Vulnerable {
    public byte[] badMD5(byte[] data) throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");
        return md.digest(data);
    }

    public byte[] badSHA1(byte[] data) throws Exception {
        MessageDigest md = MessageDigest.getInstance("SHA-1");
        return md.digest(data);
    }

    public Cipher badDES() throws Exception {
        return Cipher.getInstance("DES/ECB/PKCS5Padding");
    }

    public Cipher bad3DES() throws Exception {
        return Cipher.getInstance("DESede/ECB/PKCS5Padding");
    }

    public Cipher badRC4() throws Exception {
        return Cipher.getInstance("RC4");
    }
}

File: testdata/vulnerable.go
package vulnerable

import (
	"crypto/des"
	"crypto/md5"
	"crypto/rc4"
	"crypto/sha1"
)

func badMD5(data []byte) [16]byte {
	return md5.Sum(data)
}

func badSHA1(data []byte) [20]byte {
	return sha1.Sum(data)
}

func badDES(key []byte) {
	_, _ = des.NewCipher(key)
}

func bad3DES(key []byte) {
	_, _ = des.NewTripleDESCipher(key)
}

func badRC4(key []byte) {
	_, _ = rc4.NewCipher(key)
}
